dev/AppLifecycle/ProtocolActivatedEventArgs.h

```diff
@@ -10,21 +10,21 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
     using namespace winrt::Windows::Foundation::Collections;
     using namespace winrt::Windows::ApplicationModel::Activation;
 
-    static PCWSTR c_protocolContractId = L"Windows.Protocol";
+    constexpr PCWSTR c_protocolContractId = L"Windows.Protocol";
 
-    class ProtocolActivatedEventArgs : public winrt::implements<ProtocolActivatedEventArgs, ActivatedEventArgsBase, IProtocolActivatedEventArgs,
+    class ProtocolActivatedEventArgs : public winrt::implements<ProtocolActivatedEventArgs, IProtocolActivatedEventArgs, ActivatedEventArgsBase,
         IInternalValueMarshalable>
     {
     public:
-        ProtocolActivatedEventArgs(const std::wstring& uri) : m_uri(winrt::Windows::Foundation::Uri(uri))
+        ProtocolActivatedEventArgs(const std::wstring& uri) : m_uri(winrt::Windows::Foundation::Uri(std::move(uri)))
         {
             m_kind = ActivationKind::Protocol;
         }
 
-        static IActivatedEventArgs Deserialize(winrt::Windows::Foundation::Uri const& uri)
+        static winrt::Windows::Foundation::IInspectable Deserialize(winrt::Windows::Foundation::Uri const& uri)
         {
             auto query = uri.QueryParsed();
```

Comment:
>QueryParsed [](http://example.com/codeflow?start=29&length=11)

This is crazy. We are parsing a query string, and then reparsing it as a query string. This means that if you want to pass a percent sign all the way to the end, you have to know exactly how many levels of quoting you need to perform. `----ms-protocol:Windows.Protocol=contoso:page1?progress=35%2525` The `%2525` turns into `%25` when parsed by ExtensionContract.h, and then the `%25` turns into `%` when reparsed by ProtocolActivationEventArgs.